web/src/reload_setup.ts

```diff
@@ -27,14 +27,23 @@ const legacy_reload_vars_schema = z.intersection(
         z.object({
             msg: z.optional(z.undefined()),
         }),
-        z.object({
-            msg: z.string(),
-            recipient: z.string(),
-            message_type: z.enum(["private", "stream"]),
-            stream_id: z.optional(z.string()),
-            topic: z.optional(z.string()),
-            draft_id: z.optional(z.string()),
-        }),
+        z.intersection(
+            z.object({
+                msg: z.string(),
+                draft_id: z.optional(z.string()),
+            }),
+            z.discriminatedUnion("msg_type", [
+                z.object({
+                    msg_type: z.literal("private"),
+                    recipient: z.string(),
+                }),
+                z.object({
+                    msg_type: z.literal("stream"),
+                    stream_id: z.optional(z.string()),
+                    topic: z.optional(z.string()),
```

Comment:
`topic` seems to be required for streams (at least since 057ee6633a3db9b6bbb64f47a163ef024e2d2b2a switched it from `subject`).